main.c

```diff
@@ -303,84 +67,47 @@ int main(void)
     psa_key_policy_t policy = PSA_KEY_POLICY_INIT;
     static uint8_t pubkey[pubkey_size];
     size_t pubkey_len = 0;
-    psa_key_slot_number_t atecc608a_key_slot = 10;
+    psa_key_slot_number_t atecc608a_key_slot_device = 0;
+#ifdef REGISTER_SE
+    psa_key_slot_number_t atecc608a_key_slot_psa = 1;
+#endif
 
-    printf("**********************************************\n");
     read_serial_number();
 
-    printf("**********************************************\n");
     hash_sha256(hash_input1, sizeof(hash_input1) - 1,
                 sha256_expected_hash1, sizeof(sha256_expected_hash1));
 
-    printf("**********************************************\n");
     hash_sha256(hash_input2, sizeof(hash_input2) - 1,
                 sha256_expected_hash2, sizeof(sha256_expected_hash2));
 
     /* XXX Call this before or after registering SE? */
     status = psa_crypto_init();
     ASSERT_STATUS(status, PSA_SUCCESS);
-
+#ifdef REGISTER_SE
     /* Register SE */
     status = psa_register_secure_element(atecc608a_drv_info);
     ASSERT_STATUS(status, PSA_SUCCESS);
 
-    /* TODO Register some pre-provisioned SE slots */
-    //status = psa_register_se_slot(
-    //    psa_key_id_t id,
-    //    psa_key_slot_number_t slot,
-    //    psa_key_lifetime_t lifetime,
-    //    psa_key_type_t key_type,
-    //    size_t size,
-    //    uint8_t occupied,
-    //    int32_t owner);
-
-    /* TODO Program the device with some configuration in order to enable it to
-     * sign something? Suggest to copy Azim's for testing configuration, if
-     * needed. Read entire datasheet to find out (cover to cover). */
-
-    /* TODO Make a signature */
-    /* Come up with a test vector. Signature is randomized, so we need to
-     * verify with software. */
-#if USE_SE
-    status = atecc608a_asymmetric_sign(
-        atecc608a_key_slot, alg, &hash, sizeof(hash),
-        signature, sizeof(signature), &signature_length);
-    ASSERT_STATUS(status, PSA_SUCCESS);
-#else
-    /*
-     * Generate a volatile keypair in a volatile key slot.
-     */
-    {
-
-    status = psa_allocate_key(&sign_handle);
-    ASSERT_STATUS(status, PSA_SUCCESS);
+    /* Register first slot */
+    status = psa_register_se_slot(atecc608a_key_slot_psa, atecc608a_key_slot_device,
```

Comment:
I don't think we want to register the SE or any of its slots, yet. We just want to call the driver directly for now from the example.